### ISSUE TITLE
Refreshing Impala test UDFs/UDAs should not require a full data load

### ISSUE THREAD

**Issue by wesm** [cloudera/ibis#500]
Iteration in UDF testing is a little slow right now as a result. Let's set up a Makefile that builds and uploads the test libraries / LLVM IR to HDFS. I would like to be able to do:

```
make impala-test
py.test ibis --e2e
```

cc @megvuyyuru

**Comment by laserson** [comment]
Probably easiest to roll this into the load_data script

**Comment by laserson** [comment]
You mean that when you modify the UDF code in `testing/udf/...`, you want to build it and push it easily?

Is it more complicated than these commands?

```
cd testing/udf
make
hadoop fs -put build/.../[...].so $IBIS_TEST_DATA_HDFS_DIR/udf
cd ../..
py.test ibis --e2e
```

Either way, I think the code that deals with creating the test data and loading it would benefit from a refactor *a lot*.

**Comment by wesm** [comment]
No, it's not more complicated than that (better to use ibis to do the HDFS file management), but there needs to be an established simple workflow for developers

**Comment by megvuyyuru** [comment]
This is slightly addressed by #513. Doing `python load_test_data.py -u` or `python load_test_data.py --udf` will only make the UDFs and upload them to HDFS.

**Comment by wesm** [comment]
I think we're good, thank you!
